homeassistant/components/media_player/apple_tv.py

```diff
@@ -71,34 +71,53 @@ def async_setup_platform(hass, config, async_add_entities,
     details = pyatv.AppleTVDevice(name, host, login_id)
     session = async_get_clientsession(hass)
     atv = pyatv.connect_to_apple_tv(details, hass.loop, session=session)
-    entity = AppleTvDevice(atv, name, start_off)
+    entity = AppleTvDevice(hass, atv, name, start_off)
 
-    yield from async_add_entities([entity], update_before_add=True)
+    def on_hass_stop(event):
```

Comment:
I do not think so. Doing "stop" is basically cancel on a future and that is not thread-safe, so it must be run in the event loop. So it should be OK?